appveyor.yml

```diff
@@ -18,7 +18,7 @@ install:
 # Post-install test scripts.
 test_script:
   # run tests
-  - ps: node_modules/.bin/mocha test --timeout 4000 --R
+  - ps: node_modules/.bin/gulp ; node_modules/.bin/mocha test --timeout 4000 --R
```

Comment:
I think that would be a good improvement, but it would be good to limit this PR to a refactor. There are 7 more coming after this.